src/build/Config.zig

```diff
@@ -87,7 +88,11 @@ pub fn init(b: *std.Build) !Config {
     // This is set to true when we're building a system package. For now
     // this is trivially detected using the "system_package_mode" bool
     // but we may want to make this more sophisticated in the future.
-    const system_package: bool = b.graph.system_package_mode;
+    const system_package = b.option(
```

Comment:
For now I'd say nix the new option and continue to use `b.graph.system_package_mode`. That's what it's technically there for. 

If we need something to more carefully control the templating of the dist files we should consider whether we want to do something like this or a dedicated flag like you used to have, but I want to defer that to later.